### DISCUSSION TITLE
Source Signal Encoding as Sensor

### DISCUSSION THREAD

**Discussion by gtporter** [mvdwetering/yamaha_ynca#154] [Ideas]
Is it possible to get the source audio encoding as a sensor?   I want to be able to tell if the receiver is getting Dolby Digital, DTS, Atmos, etc. without needing to look at the display on the receiver (it is in another room).  This is a photo of what my receiver displays with Atmos content.  The second line is what I am seeking as a sensor in HA.

![PXL_20230327_212623484](https://user-images.githubusercontent.com/11232284/228635212-a200c163-7125-40a6-ab57-ef19fbc97fd4.jpg)

**Comment by mvdwetering** [comment]
I think it is unlikely that it reports Atmos etc..., but we can check.

Just make sure the receiver switches between something and Atmos and download the diagnostics by selecting "Download diagnostics" from the integration menu and attach the file here. It will have the latest info received from the receiver and we can see if something useful is there.

**Reply by gtporter** [reply]
OK, I turned on the receiver, played some DD+ Atmos content, then went back to the main menu ("Decoder Off") was displayed on the screen, and turned everything off.  Diagnostic info attached.  Thanks!

[config_entry-yamaha_ynca-c49964a760320154165a98f4e80e2bf3.json (1).txt](https://github.com/mvdwetering/yamaha_ynca/files/11125134/config_entry-yamaha_ynca-c49964a760320154165a98f4e80e2bf3.json.1.txt)

**Reply by mvdwetering** [reply]
Had a look, but there is nothing in there indicating Atmos or any other type of content, so won't be able to build a sensor for that.
